internal/fs/hns_bucket_test.go

```diff
@@ -100,3 +100,101 @@ func (t *HNSBucketTests) TestDeleteFolder() {
 	_, err = os.Stat(dirPath)
 	assert.NotNil(t.T(), err)
 }
+
+func (t *HNSBucketTests) TestRenameFolderWithSrcDirectoryDoesNotExist() {
+	oldDirPath := path.Join(mntDir, "foo_not_exist")
+	newDirPath := path.Join(mntDir, "foo_rename")
+
+	err = os.Rename(oldDirPath, newDirPath)
+
+	assert.NotNil(t.T(), err)
+	_, err = os.Stat(newDirPath)
+	assert.NotNil(t.T(), err)
+}
+
+func (t *HNSBucketTests) TestRenameFolderWithDstDirectoryIsNotEmpty() {
+	oldDirPath := path.Join(mntDir, "foo")
+	_, err = os.Stat(oldDirPath)
+	assert.NoError(t.T(), err)
+	newDirPath := path.Join(mntDir, "bar")
+	_, err = os.Stat(newDirPath)
+	assert.NoError(t.T(), err)
+
+	err = os.Rename(oldDirPath, newDirPath)
+
+	assert.NotNil(t.T(), err)
```

Comment:
nit: following is more suited.

```suggestion
	assert.Error(t.T(), err)
```